### PR TITLE
check s3URL scheme upon AWS ObjectStore Init()

### PR DESCRIPTION
**Pull request by metadave**
fixes https://github.com/heptio/ark/issues/641

### CODE REVIEW

File: pkg/cloudprovider/aws/object_store.go [hunk 1]
```diff
@@ -85,6 +85,10 @@ func (o *objectStore) Init(config map[string]string) error {
 		WithS3ForcePathStyle(s3ForcePathStyle)
 
 	if s3URL != "" {
+		if !IsValidS3URLScheme(s3URL) {
```
Comment (skriss) [inline pkg/cloudprovider/aws/object_store.go]: I think we could make this a little bit more robust by using `net/url`'s `Parse` function, which will return an error if it's not a valid URL, and then we can check the returned struct's `Scheme` property to explicitly ensure it's http/https.  WDYT?

Comment (metadave) [reply pkg/cloudprovider/aws/object_store.go]: I agree! I'll push a change shortly.

**Comment by skriss** [comment]
LGTM - please squash before we merge.  Thanks for the contribution!  

cc @nrb @carlisia for review/merge.

File: pkg/cloudprovider/aws/helpers.go [hunk 2]
```diff
@@ -51,3 +52,17 @@ func GetBucketRegion(bucket string) (string, error) {
...
+// IsValidS3URLScheme returns true if the scheme is http:// or https://
+// and the url parses correctly, otherwise, return false
+func IsValidS3URLScheme(s3URL string) bool {
+	u, err := url.Parse(s3URL)
+	if err != nil {
+		return false
+	}
+	if u.Scheme == "http" || u.Scheme == "https" {
+		return true
+	} else {
```
Comment (carlisia) [inline pkg/cloudprovider/aws/helpers.go]: A couple comments minor in consequence but with the potential to improve readability:
1. The `else` here is unnecessary.
2. I also propose to switch and say `if u.Scheme != "http" && u.Scheme != "https" ` then return `false`, lastly return `true`.

Comment (metadave) [reply pkg/cloudprovider/aws/helpers.go]: that's a good suggestion, thanks!

**Comment by skriss** [comment]
Thanks again @metadave!

**Pull request merged**